src/components/ec-table-pagination/ec-table-pagination.spec.js

```diff
@@ -0,0 +1,130 @@
+import { mount } from '@vue/test-utils';
+import EcTablePagination from './ec-table-pagination.vue';
+import { withMockedConsole } from '../../../tests/utils/console';
+
+describe('EcTablePagination', () => {
+  function mountEcTablePagination(props, mountOpts) {
+    return mount(EcTablePagination, {
+      propsData: { ...props },
+      stubs: { EcPopover: true },
+      ...mountOpts,
+    });
+  }
+
+  it('should render as expected', () => {
+    const wrapper = mountEcTablePagination();
+    expect(wrapper.element).toMatchSnapshot();
+  });
+
+  it('should throw if the prop numberOfItems has a invalid value', () => {
+    withMockedConsole((errorSpy) => {
+      mountEcTablePagination({ numberOfItems: 7 });
+      expect(errorSpy).toHaveBeenCalledTimes(1);
+      expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: custom validator check failed for prop "numberOfItems"');
+    });
+  });
+
+  it('should display page information by calculating total number of pages', () => {
+    const wrapper = mountEcTablePagination({ total: 20, numberOfItems: 5, page: 2 });
+    expect(wrapper.findByDataTest('ec-table-pagination__current-page').element).toMatchSnapshot();
+  });
+
+  it('should display last page information by calculating total number of pages', () => {
+    const wrapper = mountEcTablePagination({ total: 20, numberOfItems: 5, page: 4 });
+    expect(wrapper.findByDataTest('ec-table-pagination__current-page').element).toMatchSnapshot();
+  });
+
+  it('should display first page information by calculating total number of pages', () => {
+    const wrapper = mountEcTablePagination({ total: 20, numberOfItems: 5, page: 1 });
+    expect(wrapper.findByDataTest('ec-table-pagination__current-page').element).toMatchSnapshot();
+  });
+
+  it('should display page information even if the total is 0', () => {
+    const wrapper = mountEcTablePagination({ total: 0 });
+    expect(wrapper.findByDataTest('ec-table-pagination__current-page').element).toMatchSnapshot();
+  });
+
+  it('should display page information when last page is incomplete', () => {
+    const wrapper = mountEcTablePagination({ total: 29, numberOfItems: 5, page: 6 });
+    expect(wrapper.findByDataTest('ec-table-pagination__current-page').element).toMatchSnapshot();
+  });
+
+  it('should display the page size dropdown with preselected value', () => {
+    const wrapper = mountEcTablePagination({ numberOfItems: 50 });
+    expect(wrapper.findByDataTest('ec-table-pagination__page-size').element).toMatchSnapshot();
+  });
+
+  it('should display next and prev buttons disabled if pagination has only one page', () => {
+    const wrapper = mountEcTablePagination({ numberOfItems: 50, total: 50, page: 1 });
+    expect(wrapper.findByDataTest('ec-table-pagination__actions').element).toMatchSnapshot();
+  });
+
+  it('should display only next button disabled if pagination is on the last page', () => {
```

Comment:
personally I'd join these with a describe "when pagination is on the last page" with the same above, reusing the wrapper in a `beforeAll`

this works too 👍